ARM: Fix compile error for unw usage with clang-3.8

seanshpark opened the pull request:
This fix is also needed when using clang-3.8
Previous related patch is #4260

seanshpark commented:
@myungjoo , @janvorli , could you please review?
I could not find `CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM` and changed it to `PAL_CMAKE_PLATFORM_ARCH_ARM` like in other places.

myungjoo commented:
`CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM ` is located at /CMakeList.txt.
Is using `CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM` creating errors in LLVM 3.8?

janvorli commented:
It looks like the CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM is no longer defined in the root CMakeLists.txt or anywhere else. @rahku has renamed that to CLR_CMAKE_PLATFORM_UNIX_ARM in his recent change. But in PAL's CMakeLists.txt, we are using the PAL_CMAKE_PLATFORM_ARCH_ARM, so this change is good.

janvorli commented:
@dotnet-bot test OSX x64 Checked Build and Test please

seanshpark commented:
I've removed `exception/seh-unwind.cpp` from adding `-Wno-error=inline-asm` option, since `seh-unwind.cpp` is included from `seh.cpp` file. Thus main issue of this patch is change  `CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM` to `PAL_CMAKE_PLATFORM_ARCH_ARM`.
And also rebased to master.

janvorli commented:
LGTM
